jdaviz/configs/default/plugins/data_quality/dq_utils.py

```diff
@@ -0,0 +1,73 @@
+from importlib import resources
+from pathlib import Path
+from astropy.table import Table
+
+dq_flag_map_paths = {
+    'jwst': Path('data', 'data_quality', 'jwst.csv'),
+    'roman': Path('data', 'data_quality', 'roman.csv'),
+}
+
+
+def load_flag_map(mission_or_instrument=None, path=None):
+    """
+    Load a flag map from disk.
+
+    Parameters
+    ----------
+    mission_or_instrument : {"jwst", "roman"}, optional
+        Load the flag map for this mission or instrument.
+    path : path-like
+        Path to a flag map file (CSV).
+
+    Returns
+    -------
+    flag_mapping : dict
+        Flag mapping definitions. Keys are powers of two,
+        values are the "name" and "description" for each flag.
+    """
+    if mission_or_instrument is not None and path is None:
+        flag_map_path = dq_flag_map_paths.get(mission_or_instrument)
+
+        if flag_map_path is None:
+            raise NotImplementedError("No flag map found for "
+                                      f"mission/instrument '{mission_or_instrument}'.")
+
+    elif mission_or_instrument is None and path is not None:
+        # load directly from a CSV file:
+        flag_map_path = path
+
+    with resources.as_file(resources.files('jdaviz').joinpath(flag_map_path)) as path:
```

Comment:
Note – we use `pkg_resources` to find line list data files within jdaviz. `pkg_resources` [has been deprecated](https://setuptools.pypa.io/en/latest/pkg_resources.html), so I'm giving the recommended alternatives a go (from `importlib`).